src/core/border_router/routing_manager.cpp

```diff
@@ -574,8 +613,8 @@ void RoutingManager::EvaluateRoutingPolicy(void)
     }
 
     // 3. Update advertised on-link & OMR prefixes information.
-    mAdvertisedOnLinkPrefix = newOnLinkPrefix;
-    mAdvertisedOmrPrefixes  = newOmrPrefixes;
+    mIsAdvertisingLocalOnLinkPrefix = (newOnLinkPrefix == &mLocalOnLinkPrefix);
```

Comment:
Need to deprecate local On-Link Prefix when mIsAdvertisingLocalOnLinkPrefix is changed from true to false?

Reply:
Yes, it is done in `EvaluateOnLinkPrefix`. We update `mIsAdvertisingLocalOnLinkPrefix` here only after we have sent the PIO in `SendRouterAdvertisement`.